### PR TITLE
[Energy] Fix frequency handling

### PR DESCRIPTION
**Pull request by SteWers**
## Description:
- Fix handling of frequency
- ~~Prepare energy driver for using MQTT subscribe~~
## Checklist:
  - [x] The pull request is done against the latest development branch
  - [x] Only relevant files were touched
  - [x] Only one feature/fix was added per PR and the code change compiles without warnings
  - [x] The code change is tested and works with Tasmota core ESP8266 V.2.7.6
  - [x] The code change is tested and works with Tasmota core ESP32 V.2.0.14
  - [x] I accept the [CLA](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md#contributor-license-agreement-cla).

### CODE REVIEW

**Comment by barbudor** [comment]
You may want to check some boxes here too : 
https://github.com/arendst/Tasmota/blob/717873eac7e4f5cb29d346c6a53af497fd3ec3f9/API.md?plain=1#L41-L43

**Comment by SteWers** [comment]
Done. Thanks!

File: tasmota/tasmota_xdrv_driver/xdrv_03_esp32_energy.ino [hunk 1]
```diff
@@ -891,12 +891,11 @@ void EnergyEverySecond(void) {
           Energy->voltage[i] = 0;
           Energy->current[i] = 0;
           Energy->active_power[i] = 0;
-          if (!isnan(Energy->apparent_power[i])) { Energy->apparent_power[i] = 0; }
-          if (!isnan(Energy->reactive_power[i])) { Energy->reactive_power[i] = 0; }
-          if (!isnan(Energy->frequency[i]) && !Energy->type_dc) { Energy->frequency[i] = 0; }
-          if (!isnan(Energy->power_factor[i])) { Energy->power_factor[i] = 0; }
-          if (!isnan(Energy->export_active[i])) { Energy->export_active[i] = 0; }
-
+          if (!isnan(Energy->apparent_power[i])) Energy->apparent_power[i] = 0;
```
Comment (arendst) [inline tasmota/tasmota_xdrv_driver/xdrv_03_esp32_energy.ino:894]: Why oh why removing my favorite brackets???? Pls reinstall asap

**Comment by arendst** [comment]
I feel a lot of overhead is coming when mqtt subscribe needs to be serviced. I suggest you provide the other PR too (servicing the request) for analysis. I prefer fixing this using rules instead of hardcoding.

File: tasmota/tasmota_xdrv_driver/xdrv_03_esp32_energy.ino (hunk 1 above)
Comment (SteWers) [reply tasmota/tasmota_xdrv_driver/xdrv_03_esp32_energy.ino]: Your loved brackets are back. 😉
For me no brackets looking better in one-line-statement, but in future I'll place them.

**Comment by SteWers** [comment]
Yes, you're right. There comes one code line to the other. The PR is still in work and I'll think about another solution.

**Pull request merged**